docs/architecture/adr/0002-ai-driven-development-architecture.md

```
@@ -0,0 +1,131 @@
+# ADR-0002: Adopt AI-Driven Development Architecture
+
+## Status
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)